### PR TITLE
model:Transformers: Adding New Models

### PR DESCRIPTION
**Pull request by 0dust**
HuggingFace Transformers NER Model

### CODE REVIEW

File: model/transformers/dffml_model_transformers/ner/utils.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        else:
+            input_ids += [pad_token] * padding_length
+            input_mask += [0 if mask_padding_with_zero else 1] * padding_length
+            segment_ids += [pad_token_segment_id] * padding_length
+            label_ids += [pad_token_label_id] * padding_length
+
+        assert len(input_ids) == max_seq_length
+        assert len(input_mask) == max_seq_length
+        assert len(segment_ids) == max_seq_length
+        assert len(label_ids) == max_seq_length
```
Comment (pdxjohnny) [inline model/transformers/dffml_model_transformers/ner/utils.py]: assert is dangerous as it is removed from byte compiled python code. Let's do an `if comparison: raise AssertionError()` instead.

File: model/transformers/setup.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+INSTALL_REQUIRES = [
+    "tensorflow>=2.0.0",
+    "pandas>=0.25.0",
+    "numpy>=1.16.4",
+    "seqeval>=0.0.12",
+    "fastprogress>=0.2.2",
+    "transformers>=2.5.1",
+] + (
+    ["dffml>=0.3.4"]
```
Comment (pdxjohnny) [inline model/transformers/setup.py]:
```suggestion
    ["dffml>=0.3.5"]
```

File: model/transformers/tests/test_model.py [hunk 3]
```diff
@@ -0,0 +1 @@
+# TODO add unit tests for ner_tagger
```
Comment (pdxjohnny) [inline model/transformers/tests/test_model.py]: :+1:

File: model/transformers/dffml_model_transformers/ner/ner_model.py [hunk 4]
```diff
@@ -0,0 +1,905 @@
...
+        default=None,
+    )
+    tokenizer_name: str = field(
+        "Pretrained tokenizer name or path if not the same as model_name",
+        default=None,
+    )
+    cache_dir: str = field(
+        "Directory to store the pre-trained models downloaded from s3",
+        default=os.path.join(
+            os.path.expanduser("~"), ".cache", "dffml", "transformers"
```
Comment (pdxjohnny) [inline model/transformers/dffml_model_transformers/ner/ner_model.py]:
```suggestion
            pathlib.Path("~", ".cache", "dffml", "transformers")
```

File: model/transformers/dffml_model_transformers/ner/ner_model.py [hunk 5]
```diff
@@ -0,0 +1,905 @@
...
+        "Pretrained config name or path if not the same as model_name",
+        default=None,
+    )
+    tokenizer_name: str = field(
+        "Pretrained tokenizer name or path if not the same as model_name",
+        default=None,
+    )
+    cache_dir: str = field(
+        "Directory to store the pre-trained models downloaded from s3",
+        default=os.path.join(
```
Comment (pdxjohnny) [inline model/transformers/dffml_model_transformers/ner/ner_model.py]:
```suggestion
        default=pathlib.Path("~", ".cache", "dffml", "transformers"),
```

File: model/transformers/dffml_model_transformers/ner/ner_model.py (hunk 4 above)
Comment (pdxjohnny) [inline model/transformers/dffml_model_transformers/ner/ner_model.py]:
```suggestion
```

File: model/transformers/dffml_model_transformers/ner/ner_model.py [hunk 6]
```diff
@@ -0,0 +1,905 @@
...
+            os.path.expanduser("~"), ".cache", "dffml", "transformers"
+        ),
```
Comment (pdxjohnny) [inline model/transformers/dffml_model_transformers/ner/ner_model.py]:
```suggestion
```

File: model/transformers/dffml_model_transformers/ner/ner_model.py [hunk 7]
```diff
@@ -0,0 +1,994 @@
...
+
+
+@entrypoint("ner_tagger")
+class NERModel(Model):
+    """
+    Implemented using HuggingFace Transformers Tensorflow based Models.
+
+    First we create the training and testing datasets
+
+    .. literalinclude:: /../model/transformers/examples/ner/train_data.sh
```
Comment (pdxjohnny) [inline model/transformers/dffml_model_transformers/ner/ner_model.py]: It looks like we're missing these examples?

Comment (0dust) [reply model/transformers/dffml_model_transformers/ner/ner_model.py]: Ohh Yes!!

File: model/transformers/tests/test_ner_integration.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+            writer = csv.writer(data_file, delimiter=",")
+            writer.writerow(["A", "B"])
+            writer.writerows(PREDICT_DATA)
+
+        # Train the model
+        await CLI.cli(
+            "train",
+            "-model",
+            "ner_tagger",
+            "-model-SENTENCE_ID",
```
Comment (pdxjohnny) [inline model/transformers/tests/test_ner_integration.py]:
```suggestion
            "-model-sid",
```

Let's also make this lower case

File: model/transformers/dffml_model_transformers/ner/ner_model.py [hunk 9]
```diff
@@ -0,0 +1,994 @@
...
+        RobertaConfig,
+        TFRobertaForTokenClassification,
+        RobertaTokenizer,
+    ),
+}
+
+
+@config
+class NERModelConfig:
+    SENTENCE_ID: Feature = field(
```
Comment (pdxjohnny) [inline model/transformers/dffml_model_transformers/ner/ner_model.py]:
```suggestion
    sid: Feature = field(
```

File: model/transformers/dffml_model_transformers/ner/ner_model.py [hunk 10]
```diff
@@ -0,0 +1,994 @@
...
+        "Unique Id to identify words of each sentence"
```
Comment (pdxjohnny) [inline model/transformers/dffml_model_transformers/ner/ner_model.py]:
```suggestion
        "Unique Id to identify words of each sentence (Sentence ID)"
```

Comment (pdxjohnny) [reply model/transformers/dffml_model_transformers/ner/ner_model.py]: just so the acronym (sid) is clear

**Pull request merged**